bids_conversion/convert_somato_data.py

```diff
@@ -0,0 +1,270 @@
+from mne.utils import check_version, run_subprocess
+import mne.datasets.somato as somato
+from mne_bids import write_raw_bids, make_bids_basename, write_anat
+from mne_bids.utils import print_dir_tree
+
+# mne should be 0.18 minimum ... but not higher
+if not check_version('mne', '0.18') or check_version('mne', '0.19'):
+    raise RuntimeError('You need to install mne 0.18: `pip install mne==0.18`')
+# the other dependencies are a bit more lenient, we check anyways
+for pkg, ver in [('numpy', '1.16'), ('scipy', '1.2'), ('nibabel', '2.4.1'),
+                 ('mne_bids', '0.2')]:  # noqa: E501 XXX: mne_bids should be changed to 0.3. Not released yet
+    if not check_version(pkg, ver):
+        raise RuntimeError('You need to install {0} {1} or higher:'
+                           'pip install {0}>={1}'.format(pkg, ver))
+
+# Check that freesurfer is initialized
+FREESURFER_HOME = os.getenv('FREESURFER_HOME')
+SUBJECTS_DIR = os.getenv('SUBJECTS_DIR')
+if FREESURFER_HOME is None or SUBJECTS_DIR is None:
+    raise RuntimeError('You need to define a FREESURFER_HOME environment '
+                       'variable pointing to your installation of freesurfer, '
+                       'and a SUBJECTS_DIR environment variable pointing to a '
+                       'folder where to save the freesurfer subject data. '
+                       'Please refer to the module docstring for help.')
+
+# Somato data prior to conversion
+somato_path = somato.data_path()
+
+# Print the directory tree
+print('\nSomato data before conversion:\n')
+print_dir_tree(somato_path, max_depth=3)
+print('\n\n')
+
+# Convert to BIDS
+# Path to the raw data of somato dataset
+fif_path = op.join(somato_path, 'MEG', 'somato')
+
+# Load the raw data file
+raw = mne.io.read_raw_fif(op.join(fif_path, 'sef_raw_sss.fif'))
+
+# Set the sex of the participant to male
+# Subject sex (0=unknown, 1=male, 2=female).
+raw.info['subject_info']['sex'] = raw.info['subject_info'].get('sex', 1)
+
+# Get the events, and assign an ID
+events = mne.find_events(raw)
+event_id = dict()
+for val in np.unique(events[:, -1]):
+    event_id['somato_event{}'.format(val)] = val
+
+print('\n\n')
+print('event_id: "{}"'.format(event_id))
+
+# Make a bids name for the raw file
+bids_basename = make_bids_basename(subject='01', task='somato')
+
+# Make a location for the BIDS formatted data
+somato_parent, somato_name = op.split(somato_path)
+somato_path_bids = op.join(somato_parent, somato_name + '-bids')
+
+# Write the data
+write_raw_bids(raw, bids_basename, somato_path_bids, events, event_id,
+               overwrite=True)
+
+# Edit dataset_description.json to include link to dataset
+dataset_description_json = op.join(somato_path_bids,
+                                   'dataset_description.json')
+with open(dataset_description_json, 'r') as fin:
+    jsondict = json.load(fin)
+
+# Edit some fields we know of
+jsondict['Name'] = 'MNE-somato-data-bids'
+jsondict['Authors'] = ['Lauri Parkkonen']
```

Comment:
Are we just reformatting it or putting it on OpenNeuro as well? If you want to put it on OpenNeuro, we should ask @LauriParkkonen

Reply:
I thought reformatting and putting on OSF as a new version of the same data (version 4): https://osf.io/tp4sg/

but Lauri can say whether he is fine with the License that we are giving the data now:

'PDDL, see: https://opendatacommons.org/licenses/pddl/'

Alex told me I should put a permissive one :-)